components/src/lib/Modal.tsx

```
@@ -0,0 +1,49 @@
+const Modal = ({ modalType, titleText = '', children, closeType, onClose }: ModalProps) => {
+  const context = useContext(ModalContext);
+  const { closeModalHandler } = useModal();
+
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      closeModalHandler();
+    }
+  };
+
+  useEffect(() => {
+    if (context && context.isModalOpened) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)